tests/filecheck/transforms/csl-stencil-to-csl-wrapper.mlir

```diff
@@ -56,38 +56,44 @@ builtin.module {
 // CHECK-NEXT:     %25 = arith.ori %24, %22 : i1
 // CHECK-NEXT:     "csl_wrapper.yield"(%14, %13, %25) <{"fields" = ["memcpy_params", "stencil_comms_params", "isBorderRegionPE"]}> : (!csl.comptime_struct, !csl.comptime_struct, i1) -> ()
 // CHECK-NEXT:   }, {
-// CHECK-NEXT:   ^1(%26 : i16, %27 : i16, %28 : i16, %29 : i16, %30 : i16, %31 : i16, %32 : i16, %memcpy_params : !csl.comptime_struct, %stencil_comms_params : !csl.comptime_struct, %isBorderRegionPE : i1, %a : !stencil.field<[-1,1023]x[-1,511]xtensor<512xf32>>, %b : !stencil.field<[-1,1023]x[-1,511]xtensor<512xf32>>):
+// CHECK-NEXT:   ^1(%26 : i16, %27 : i16, %28 : i16, %29 : i16, %30 : i16, %31 : i16, %32 : i16, %memcpy_params : !csl.comptime_struct, %stencil_comms_params : !csl.comptime_struct, %isBorderRegionPE : i1):
 // CHECK-NEXT:     %33 = "csl_wrapper.import"(%memcpy_params) <{"module" = "<memcpy/memcpy>", "fields" = [""]}> : (!csl.comptime_struct) -> !csl.imported_module
 // CHECK-NEXT:     %34 = "csl_wrapper.import"(%29, %31, %stencil_comms_params) <{"module" = "stencil_comms.csl", "fields" = ["pattern", "chunkSize", ""]}> : (i16, i16, !csl.comptime_struct) -> !csl.imported_module
+// CHECK-NEXT:     "memref.global"() <{"sym_name" = "a", "type" = memref<512xf32>, "initial_value", "sym_visibility" = "public"}> : () -> ()
+// CHECK-NEXT:     %35 = memref.get_global @a : memref<512xf32>
+// CHECK-NEXT:     %a = builtin.unrealized_conversion_cast %35 : memref<512xf32> to !stencil.field<[-1,1023]x[-1,511]xtensor<512xf32>>
```

Comment:
I don't quite follow how this `unrealized_conversion_cast`  gets passed to the `stencil.load` below. Is that done with the `arg_op_mapping`?

Reply:
Any `stencil.load` should be resolved by `--stencil-bufferize` and not be there at this stage, and everything should be based on `stencil.field<AxBxtensor<Cxf32>>`. This in turn is translated to a memref in `--csl-stencil-bufferize`, which takes the element_type (tensor) and translates it to a memref. Maybe it should translate to `memref<1x1xCxf32>` instead of `memref<Cxf32>`, but this is all linked to DMP which has changes in the pipeline that allow it to operate on `stencil.field`, so it's best to wait for this to be merged.